cli/src/pcluster/resources/imagebuilder/custom_component_script_https.yaml

```diff
@@ -0,0 +1,49 @@
+name: ScriptComponent
+description: ScriptComponent
+schemaVersion: 1.0
+
+phases:
+  - name: build
+    steps:
+      # Exit code for failures
+      - name: Fail
+        action: ExecuteBash
+        inputs:
+          commands:
+            - echo 1
+
+      # Script name
+      - name: ScriptName
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              echo {{script-name}}
+
+      # Script url
+      - name: ScriptUrl
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              echo {{script-url}}
+
+      # Download bash script
+      - name: Download
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              curl --retry 3 -L -o /tmp/{{ build.ScriptName.outputs.stdout }} {{ build.ScriptUrl.outputs.stdout }}
```

Comment:
we can simplify the logic here.
I'm not sure having the script name passed from outside is useful here. We can just download the file from `ScriptUrl` and put it in a temporary file, or if we really want the name we can parse it from the `ScriptUrl ` in a separate action.

For example we can have
```
      - name: TempScript
        action: ExecuteBash
        inputs:
          commands:
            - |
              set -v
              temp_file=$(mktmep --tmpdir /tmp/parallelcluster/)
              echo "${temp_file}

      - name: DownloadScript
        action: ExecuteBash
        inputs:
          commands:
            - |
              set -v
              curl --retry 3 -L -o {{ build.TempScript.outputs.stdout }} {{ build.ScriptUrl.outputs.stdout }}

Reply:
done